lib/lambda_ethereum_consensus/state_transition/operations.ex

```diff
@@ -375,6 +375,54 @@ defmodule LambdaEthereumConsensus.StateTransition.Operations do
     end
   end
 
+  @spec process_deposit(BeaconState.t(), SszTypes.Deposit.t()) ::
+          {:ok, BeaconState.t()} | {:error, String.t()}
+  def process_deposit(state, deposit) do
+    deposit_data_root = Ssz.hash_tree_root(deposit.data)
+
+    case deposit_data_root do
+      {:ok, deposit_data_root} ->
+        is_valid =
+          Predicates.is_valid_merkle_branch(
+            deposit_data_root,
+            deposit.proof,
+            Constants.deposit_contract_tree_depth() + 1,
+            state.eth1_deposit_index,
+            state.eth1_data.deposit_root
+          )
+
+        if is_valid do
+          apply_deposit(state, deposit)
+        else
+          {:error, "Merkle branch is not valid"}
+        end
+
+      {:error, msg} ->
+        {:error, msg}
+    end
+  end
+
+  defp apply_deposit(state, deposit) do
+    state = %BeaconState{
+      state
+      | eth1_deposit_index: state.eth1_deposit_index + 1
+    }
+
+    res =
+      Mutators.apply_deposit(
+        state,
+        deposit.data.pubkey,
+        deposit.data.withdrawal_credentials,
+        deposit.data.amount,
+        deposit.data.signature
+      )
+
+    case res do
+      {:ok, state} -> {:ok, state}
+      {:error, msg} -> {:error, msg}
+    end
```

Comment:
This case does nothing. It just returns what it received. We should:

```elixir
state
|> Map.put(:eth1_deposit_index, state.eth1_deposit_index + 1)
|> Mutators.apply_deposit(deposit.data.pubkey, ...)
```

With a bigger refactor in mind we probably wouldn't need this function at all.